packages/terra-button-group/docs/README.md

```diff
@@ -19,12 +21,85 @@ import ButtonGroup from 'terra-button-group';
     <ButtonGroup.Button text="Button 1" key="button1" />
     <ButtonGroup.Button text="Button 2" key="button2" />
   </ButtonGroup>
+```
 
-// Selectable (toggle style) button group with second button pre-selected
-  <ButtonGroup isSelectable>
-    <ButtonGroup.Button text="Button 1" key="button1" />
-    <ButtonGroup.Button isSelected text="Button 2" key="button2" />
-  </ButtonGroup>
+### Single-Select Button Group
+```jsx
+import React from 'react';
+import ButtonGroup from '../../src/ButtonGroup';
+
+class SingleSelectButtonGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { selectedKey: ['1'] };
+    this.handleSelection = this.handleSelection.bind(this);
+  }
+
+  handleSelection(event, selectedKey) {
+    if (ButtonGroup.Utils.shouldHandleSingleSelection(this.state.selectedKey, selectedKey)) {
+      event.preventDefault();
+      this.setState({ selectedKey: [selectedKey] });
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <h3>Selected Button: <span id="selected-key">{this.state.selectedKey}</span></h3>
+        <ButtonGroup
+          id="button-group-single-select"
+          onChange={this.handleSelection}
+          selectedKeys={this.state.selectedKey}
+        >
+          <ButtonGroup.Button text="Single-Select 1" key="1" />
+          <ButtonGroup.Button text="Single-Select 2" key="2" />
+          <ButtonGroup.Button text="Single-Select 3" key="3" />
+          <ButtonGroup.Button text="Single-Select 4" key="4" />
+        </ButtonGroup>
+      </div>
+    );
+  }
+}
+export default SingleSelectButtonGroup;
```

Comment:
This is just up to your preference, I find simplified examples easier to read and understand. Here is a potential example that has the bare minimum needed to work.
```
import React from 'react';
import ButtonGroup from 'terra-button-group';

class RadioButtonGroup extends React.Component {
  constructor(props) {
    super(props);
    this.state = { selectedKey: '1' };
    this.handleSelection = this.handleSelection.bind(this);
  }

  handleSelection(event, selectedKey) {
    this.setState({ selectedKey });
  }

  render() {
    return (
      <div>
        <ButtonGroup onChange={this.handleSelection} selectedKeys={[this.state.selectedKey]}>
          <ButtonGroup.Button text="Single-Select 1" key="1" />
          <ButtonGroup.Button text="Single-Select 2" key="2" />
          <ButtonGroup.Button text="Single-Select 3" key="3" />
          <ButtonGroup.Button text="Single-Select 4" key="4" />
        </ButtonGroup>
      </div>
    );
  }
}
export default RadioButtonGroup;
```
I personally think the added helper logic in the if block to conditionally set state adds more complexity and makes it harder to understand without increasing much value / performance. 

A consumer had an issue not too long ago with their logic inside their List onChange function never fired because they had copy pasted the [example blocks](https://github.com/cerner/terra-core/blob/master/packages/terra-list/docs/Selectable.md) without understanding the utils would ignore additional clicks to an already selected item.

Reply:
I just copied it from the test. Examples are now slimmed down: https://github.com/cerner/terra-core/pull/1155/commits/c4e0000d6ceaf96b85053f0046154de1799e7072